chia/wallet/vault/vault_wallet.py

```diff
@@ -0,0 +1,606 @@
+            vault_inner_solution,
+        )
+
+        vault_spend = make_spend(self.vault_info.coin, full_puzzle, full_solution)
+        all_spends = [*p2_spends, vault_spend]
+        spend_bundle = SpendBundle(all_spends, G2Element())
+
+        amount = uint64(sum([payment.amount for payment in primaries]))
+        target_puzzle_hash = primaries[0].puzzle_hash
+
+        tx_record = TransactionRecord(
+            confirmed_at_height=uint32(0),
+            created_at_time=uint64(int(time.time())),
+            to_puzzle_hash=target_puzzle_hash,
+            amount=amount,
+            fee_amount=fee,
+            confirmed=False,
+            sent=uint32(0),
+            spend_bundle=spend_bundle,
+            additions=spend_bundle.additions(),
+            removals=spend_bundle.removals(),
+            wallet_id=self.id(),
+            sent_to=[],
+            memos=[],
+            trade_id=None,
+            type=uint32(TransactionType.OUTGOING_TX.value),
+            name=spend_bundle.name(),
+            valid_times=parse_timelock_info(tuple()),
+        )
+        await self.wallet_state_manager.add_pending_transactions([tx_record], sign=False)
+        return [tx_record]
+
+    def puzzle_for_pk(self, pubkey: G1Element) -> Program:
+        raise NotImplementedError("vault wallet")
+
+    async def puzzle_for_puzzle_hash(self, puzzle_hash: bytes32) -> Program:
+        raise NotImplementedError("vault wallet")
+
+    async def sign_message(self, message: str, puzzle_hash: bytes32, mode: SigningMode) -> Tuple[G1Element, G2Element]:
+        raise NotImplementedError("vault wallet")
+
+    async def get_puzzle_hash(self, new: bool) -> bytes32:
+        if new:
+            return await self.get_new_puzzlehash()
+        else:
+            record: Optional[
+                DerivationRecord
+            ] = await self.wallet_state_manager.get_current_derivation_record_for_wallet(self.id())
+            if record is None:
+                return await self.get_new_puzzlehash()
+            return record.puzzle_hash
+
+    async def apply_signatures(
+        self, spends: List[Spend], signing_responses: List[SigningResponse]
+    ) -> SignedTransaction:
+        raise NotImplementedError("vault wallet")
+
+    async def execute_signing_instructions(
+        self, signing_instructions: SigningInstructions, partial_allowed: bool = False
+    ) -> List[SigningResponse]:
+        raise NotImplementedError("vault wallet")
+
+    async def path_hint_for_pubkey(self, pk: bytes) -> Optional[PathHint]:
+        raise NotImplementedError("vault wallet")
+
+    async def sum_hint_for_pubkey(self, pk: bytes) -> Optional[SumHint]:
+        raise NotImplementedError("vault wallet")
+
+    def make_solution(
+        self,
+        primaries: List[Payment],
+        conditions: Tuple[Condition, ...] = tuple(),
+        fee: uint64 = uint64(0),
+    ) -> Program:
+        raise NotImplementedError("vault wallet")
```

Comment:
This will have to be implemented for the sake of other wallets that are building spends with this as an inner puzzle.